rascal-vscode-extension/src/lsp/RascalLSPConnection.ts

```diff
@@ -72,172 +72,24 @@ export async function activateLanguageClient(
-                { webviewPort: 9155, extensionHostPort: 9155},
-                { webviewPort: 9156, extensionHostPort: 9156},
-                { webviewPort: 9157, extensionHostPort: 9157},
-                { webviewPort: 9158, extensionHostPort: 9158},
-                { webviewPort: 9159, extensionHostPort: 9159},
-                { webviewPort: 9160, extensionHostPort: 9160},
-                { webviewPort: 9161, extensionHostPort: 9161},
-                { webviewPort: 9162, extensionHostPort: 9162},
-                { webviewPort: 9163, extensionHostPort: 9163},
-                { webviewPort: 9164, extensionHostPort: 9164},
-                { webviewPort: 9165, extensionHostPort: 9165},
-                { webviewPort: 9166, extensionHostPort: 9166},
-                { webviewPort: 9167, extensionHostPort: 9167},
-                { webviewPort: 9168, extensionHostPort: 9168},
-                { webviewPort: 9169, extensionHostPort: 9169},
-                { webviewPort: 9170, extensionHostPort: 9170},
-                { webviewPort: 9171, extensionHostPort: 9171},
-                { webviewPort: 9172, extensionHostPort: 9172},
-                { webviewPort: 9173, extensionHostPort: 9173},
-                { webviewPort: 9174, extensionHostPort: 9174},
-                { webviewPort: 9175, extensionHostPort: 9175},
-                { webviewPort: 9176, extensionHostPort: 9176},
-                { webviewPort: 9177, extensionHostPort: 9177},
-                { webviewPort: 9178, extensionHostPort: 9178},
-                { webviewPort: 9179, extensionHostPort: 9179},
-                { webviewPort: 9180, extensionHostPort: 9180},
-                { webviewPort: 9181, extensionHostPort: 9181},
-                { webviewPort: 9182, extensionHostPort: 9182},
-                { webviewPort: 9183, extensionHostPort: 9183},
-                { webviewPort: 9184, extensionHostPort: 9184},
-                { webviewPort: 9185, extensionHostPort: 9185},
-                { webviewPort: 9186, extensionHostPort: 9186},
-                { webviewPort: 9187, extensionHostPort: 9187},
-                { webviewPort: 9188, extensionHostPort: 9188},
-                { webviewPort: 9189, extensionHostPort: 9189},
-                { webviewPort: 9190, extensionHostPort: 9190},
-                { webviewPort: 9191, extensionHostPort: 9191},
-                { webviewPort: 9192, extensionHostPort: 9192},
-                { webviewPort: 9193, extensionHostPort: 9193},
-                { webviewPort: 9194, extensionHostPort: 9194},
-                { webviewPort: 9195, extensionHostPort: 9195},
-                { webviewPort: 9196, extensionHostPort: 9196},
-                { webviewPort: 9197, extensionHostPort: 9197},
-                { webviewPort: 9198, extensionHostPort: 9198},
-                { webviewPort: 9199, extensionHostPort: 9199},
-            ]
-        }
-    );
+    vscode.env.asExternalUri(vscode.Uri.parse(url)).then(freshURL => {
```

Comment:
in typescript you can use `await` in instead of having to chain it with callbacks (it's syntatic sugar for callbacks code).

so that would make it `const freshUrl = await vscode.env.asExternalUri();`

Reply:
and just for my understanding: in normal setup, this just returns the same url as the original?

Reply:
Yes it is the identity when the client host is the same as the server host. And also then it does not have the side effect of setting up tunnels.

Reply:
I rewrote it but did not test it. testing is rather cumbersome.